### PR TITLE
Data types and number

### PR DESCRIPTION
**Pull request by Phantomcheg**

### CODE REVIEW

File: Exercises/data-types-overview/data-types-overview.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+const constructString = new String('text');
+const constructBool = new Boolean(true);
+const constructArr = new Array;
+const constructObj = new Object();
+
+// 2. Use operator typeof on each of them.
+
+const dataTypes = [ someNumber, someString, someBool, someEmpty, someUnknown, someObj, someSymbol ];
+
+dataTypes.forEach((item) => {
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js:21]: 👍

File: Exercises/data-types-overview/data-types-overview.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+// 1. Declare an array containing different data types.
+
+const arr = [ 'test1', 6, Symbol('lol'), true, {}, 18, 'anyText', 'test2', false, null, undefined ];
+
+// 2. Iterate through it and collect amount of each data type to object.
+
+const finalObject = {};
+
+arr.forEach((arrayItem) => {
+  finalObject[typeof arrayItem] = (finalObject[typeof arrayItem] || null) + 1;
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: dangerous type conversion here! you are doing `|| null` and lucky for you it casts to `0` but you may got `NaN` for example. just do obviously `|| 0`

File: Exercises/numbers/numbers.js [hunk 3]
```diff
@@ -1,3 +1,67 @@
...
+
+const roundedRandomDecimal = Math.round(randomNumber(1, 100));
+console.log(roundedRandomDecimal);
+
+// # Additional
+// 1. Make function which will take a array with numbers as a parameter and
+// calculate average number and round to hundredths.
+// 2. Generate array to pass to function above with values, and length generated by Math.random.
+
+function workWithArrays(anyArray) {
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: add more meaningful names for functions `getAverage` for example. `anyArray` also not quite good, just `array` will be enough.

File: Exercises/numbers/numbers.js [hunk 4]
```diff
@@ -1,3 +1,67 @@
...
+  let summary = null;
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: same issue as above: just `let summary = 0`

File: Exercises/numbers/numbers.js [hunk 5]
```diff
@@ -1,3 +1,67 @@
...
+  anyArray.forEach((arrayItem) => {
+    summary += arrayItem;
+  });
+  return parseFloat(summary).toFixed(3);
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: it is sum, but not average

File: Exercises/numbers/numbers.js [hunk 6]
```diff
@@ -1,3 +1,67 @@
...
+}
+
+const arrayForCalc = [];
+for (let i = 0; i <= 5; i++) {
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: make length random as well :)

File: Exercises/numbers/numbers.js [hunk 7]
```diff
@@ -1,3 +1,67 @@
...
+  arrayForCalc.push(Math.random()*10);
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: don't forgot to run `npm run lint` and fix problems. Spaces are missed

Comment (lizarusi) [inline Exercises/numbers/numbers.js]: let's complicate a task, generate random hex numbers, and in function above parse them to decimal value

**Review by lizarusi — CHANGES_REQUESTED** [review]
everything is perfect, small notes added

**Review by Mike-Savin — CHANGES_REQUESTED** [review]
See Liza's comments

**Comment by Phantomcheg** [comment]
Pushed changes, which refer to remarks you made

File: Exercises/numbers/numbers.js [hunk 8]
```diff
@@ -1,3 +1,68 @@
...
+console.log(roundedRandomDecimal);
+
+// # Additional
+// 1. Make function which will take a array with numbers as a parameter and
+// calculate average number and round to hundredths.
+// 2. Generate array to pass to function above with values, and length generated by Math.random.
+
+function averageNumberFromArray(array) {
+  let summary = 0;
+  array.forEach((arrayItem) => {
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js:56]: you can also do it with `reduce`, you can check if you want :) https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

File: Exercises/numbers/numbers.js [hunk 9]
```diff
@@ -1,3 +1,68 @@
...
+    summary += parseInt(arrayItem, 16);
+  });
+  return (summary/array.length).toFixed(3);
+}
+
+const arrayForCalc = [];
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js:62]: better not to cut variables name, so better name will be `arrayForCalculation`

**Review by lizarusi — APPROVED** [review]
Small notes added, but in general perfect :)

File: Exercises/numbers/numbers.js [hunk 10]
```diff
@@ -1,3 +1,68 @@
...
+const hexFromDecimal = someDecimalNumber.toString(16);
+console.log(hexFromDecimal);
+
+// 4. use `Math.random` to generate random value from `0` to `100`.
+
+function randomNumber(min, max) {
+  return min + Math.floor(Math.random() * (max - min + 1));
+}
+
+const roundedRandomDecimal = Math.round(randomNumber(1, 100));
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js:46]: By the way, Math.round here also can be omitted

File: Exercises/numbers/numbers.js [hunk 11]
```diff
@@ -1,3 +1,68 @@
...
+// 1. Make function which will take a array with numbers as a parameter and
+// calculate average number and round to hundredths.
+// 2. Generate array to pass to function above with values, and length generated by Math.random.
+
+function averageNumberFromArray(array) {
+  let summary = 0;
+  array.forEach((arrayItem) => {
+    summary += parseInt(arrayItem, 16);
+  });
+  return (summary/array.length).toFixed(3);
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js:59]: It's better to use spaces around operators

**Review by Mike-Savin — APPROVED** [review]
Looks good 👍, just see our comments for the future

**Pull request merged**